se_core/include/se/node_iterator.hpp

```diff
@@ -103,18 +103,26 @@ class node_iterator {
     return occupiedVoxels;
   }
 
-  vec3i getFreeVoxels(float threshold , const uint64_t morton) {
+  vec3i getFreeVoxels(float threshold, const uint64_t morton) {
     vec3i freeVoxels;
     freeVoxels.clear();
 
     typename VoxelBlock<T>::value_type value;
-    Eigen::Vector3i blockCoord = unpack_morton(morton);
-
-    VoxelBlock<T>* block = map_.fetch(morton);
+    const Eigen::Vector3i blockCoord = keyops::decode(morton);
+    const int level = keyops::level(morton);
+
+    Node<T> *node = nullptr;
+    bool is_block = false;
+    map_.fetch_octant(blockCoord(0), blockCoord(1), blockCoord(2), node, is_block);
+    if (!is_block) {
```

Comment:
It would be easier to follow if the if and else were swapped so that the test is `if (is_block)`.